[pickers] Remove automatic switch between date and time or between range positions (#17166)
mui/mui-x: docs/data/migration/migration-pickers-v7/migration-pickers-v7.md

## Components breaking changes

### ⚠️ Updated view selection process

The view selection process has been updated to be more clear across all Pickers.
Only `<DesktopDatePicker />` and `<DesktopDateRangePicker />` maintain the previous behavior of closing after the selection is complete and switching to end range position when start value is selected.
In essence, the automatic range position and view switching has been removed in favor of manual confirmation.
The new default behavior for all other cases is as follows:

- Selection on a given view has to be confirmed by clicking "**Next**" action button if there are other selection steps.
- The "**Next**" action is replaced with ""**OK**" if there is no next step.
- The ""**OK**" action has to be clicked to confirm the selection and close the Picker.

Here are a few examples of how the new behavior works:

- On `<DesktopDateTimePicker />`:

  - Previously selecting a date and selecting all time sections automatically closed the Picker.
  - Now, the user has to click "**OK**" to confirm the selection and close the Picker regardless of the selection process.

> - On `<MobileDateTimeRangePicker />`:

Reviewer: Added a section: https://github.com/mui/mui-x/pull/17166/commits/529a8dc781f2b4244666402de536e47ea36ef56d 😉

## Components breaking changes

### ⚠️ Updated view selection process

> The view selection process has been updated to be more clear across all Pickers.

Reviewer: ```suggestion
The view selection process has been updated to make it clearer across all Pickers.
```

> Only `<DesktopDatePicker />` and `<DesktopDateRangePicker />` maintain the previous behavior of closing after the selection is complete and switching to end range position when start value is selected.

Reviewer: ```suggestion
Only `<DesktopDatePicker />` and `<DesktopDateRangePicker />` maintain the previous behavior of closing after the selection is complete and switching to the end range position when the start value is selected.
```

> In essence, the automatic range position and view switching has been removed in favor of manual confirmation.

Reviewer: ```suggestion
In essence, the automatic range position and view switching have been removed in favor of manual confirmation.
```

The new default behavior for all other cases is as follows:

> - Selection on a given view has to be confirmed by clicking "**Next**" action button if there are other selection steps.

Reviewer: ```suggestion
- Selection on a given view has to be confirmed by clicking the "**Next**" action button if there are other selection steps.
```

- The "**Next**" action is replaced with ""**OK**" if there is no next step.

> - The ""**OK**" action has to be clicked to confirm the selection and close the Picker.

Reviewer: ```suggestion
- The "**OK**" action has to be clicked to confirm the selection and close the Picker.
```

Here are a few examples of how the new behavior works:

- On `<DesktopDateTimePicker />`:

  - Previously selecting a date and then selecting all time sections automatically closed the Picker.
  - Now, the user has to click "**OK**" to confirm the selection and close the Picker regardless of the selection process.

- On `<DesktopDateTimeRangePicker />`:

>   - Previously selecting a start date and then selecting all time sections automatically switched to end date selection step. After the last end time section was selected, the Picker closed.

Reviewer: ```suggestion
  - Previously selecting a start date and then selecting all time sections automatically switched to the end date selection step. After the last end time section was selected, the Picker closed.
```

  - Now, the user has to click "**Next**" to confirm the start date and time selection to get to the end date and time selection step. Clicking "**OK**" while on the end date and time step confirms the selection and closes the Picker.

- On `<MobileDateTimeRangePicker />`:

>   - Previously selecting a start date automatically switched to start time selection step. After the last time section was selected, the Picker switched to the end date step.

Reviewer: ```suggestion
  - Previously selecting a start date automatically switched to the start time selection step. After the last time section selection, the Picker switched to the end date step.
```
